### PR TITLE
Color mode selector for css files.

### PR DESCRIPTION
**Pull request by lukasoppermann**
## Summary

This adds a selector like `:root, [data-color-mode="light"][data-light-theme="light"]` to the css files instead of just `:root`

## List of notable changes:

- **added** `getCssSelector` function to css platform to determine selector based on output filename
- **added** `css/themed` formatter to output css file with `prefers-color-scheme`
- **updated** `css` platform to include `options.selector = getCssSelector(outputFile)`

This produces the following for `dark-dimmed.css` and other `dark-...` files:

```css
:root,
[data-color-mode="dark"][data-dark-theme="dark_dimmed"] {
    //...
}
@media (prefers-color-scheme: light) {
  [data-color-mode="auto"][data-light-theme="dark_dimmed"] {
    //...
  }
}
@media (prefers-color-scheme: dark) {
  [data-color-mode="auto"][data-dark-theme="dark_dimmed"] {
    //...
  }
}
```

And the following for all `light-` css files.

```css
:root,
[data-color-mode="light"][data-dark-theme="light"] {
    //...
}
@media (prefers-color-scheme: light) {
  [data-color-mode="auto"][data-light-theme="light"] {
    //...
  }
}
@media (prefers-color-scheme: dark) {
  [data-color-mode="auto"][data-dark-theme="light"] {
    //...
  }
}
```

## What should reviewers focus on?

- is this what was desire in https://github.com/github/primer/issues/1408

## Steps to test:

1. clone and install branch
1. run `npm run build:color-tokens`
1. review output color files in `tokens-v2-private/css`

## Contributor checklist:

- [x] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change
- [ ] Verify the design tokens changed in this PR are expected using the diffing results in this PR

### CODE REVIEW

**Comment by lukasoppermann** [comment]
As discussed with @langermank and @jonrohan we probably need `prefers-color-scheme` support.
I added a new formatter `css/themed` to provided this.

**Review by langermank — APPROVED** [review]
I think this looks correct! I'd love for @jonrohan to also run the build and check the output 🙌

File: config/platforms/css.ts [hunk 1]
```diff
@@ -2,21 +2,41 @@ import StyleDictionary from 'style-dictionary'
...
-      destination: `${outputFile}`,
-      format: `css/variables`,
-      filter: isSource,
-      options: {
-        outputReferences: false
+const getCssSelectors = (outputFile: string): {selector: string; selectorLight: string; selectorDark: string} => {
+  // check for dark in the beginning of the output filename
+  const lastSlash = outputFile.lastIndexOf('/')
+  const outputBasename = outputFile.substring(lastSlash + 1, outputFile.indexOf('.'))
+  const themeName = outputBasename.replace('-', '_')
```
Comment (langermank) [inline config/platforms/css.ts:9]: Noticed one small bug testing this with @jonrohan– this only seems to replace the first `-`, but the whole theme name should use underscores like `dark_high_contrast`. We tried to fix this, but ran into typescript issues and couldn't solve it 😅

Comment (lukasoppermann) [reply config/platforms/css.ts:9]: Ahh, good catch, will fix it.

**Pull request merged**